src/esp32/i2c.rs

```diff
@@ -0,0 +1,187 @@
+#![allow(dead_code)]
+
+use crate::common::config::{AttributeError, Kind};
+use crate::common::i2c::I2CHandle;
+use esp_idf_hal::delay::BLOCK;
+use esp_idf_hal::gpio::AnyIOPin;
+use esp_idf_hal::i2c::{I2cConfig, I2cDriver};
+use esp_idf_hal::peripherals::Peripherals;
+use esp_idf_hal::units::Hertz;
+
+#[derive(Copy, Clone, Debug)]
+pub struct Esp32I2cConfig {
+    pub name: &'static str,
+    pub bus: &'static str,
+    pub baudrate_hz: u32,
+    pub timeout_ns: u32,
+    pub data_pin: i32,
+    pub clock_pin: i32,
+}
+
+impl From<Esp32I2cConfig> for I2cConfig {
+    fn from(value: Esp32I2cConfig) -> I2cConfig {
+        // TODO: when next version of esp_idf_hal is released, use below instead
```

Comment:
is it covered in the bumped version we just merged?

Reply:
No the version we bumped to (the latest) was released in December and the change allowing timeout configuration was committed in February